packages/completer/src/handler.ts

```diff
@@ -192,8 +192,8 @@ export class CompletionHandler implements IDisposable {
     }
 
     const { start, end, value } = patch;
-    editor.model.value.remove(start, end);
-    editor.model.value.insert(start, value);
+    // we need to update the shared model in a single transaction so that the undo manager works as expected
+    editor.model.sharedModel.updateSource(start, end, value);
```

Comment:
This change appears to introduce a bug with the cursor moving to the beginning of the token rather to the end when selecting completion items. As it breaks the UX for all users (not only user using RTC) I consider it to be blocking. It is already tracked in https://github.com/jupyter-rtc/jupyterlab/issues/64

Reply:
Thanks for tracking this down! This will be fixed by https://github.com/jupyterlab/jupyterlab/pull/10234